File: projecteuler/euler005.rb
MAX = 41

def self.esieve
  @esieve ||= begin
    sieve = Array.new(MAX) { 0 }
    sieve[0] = 1
    sieve[1] = 1

    i = 2
    while i < MAX do
      if sieve[i] == 0
        j = i * i
        while j < MAX do
          sieve[j] = 1
          j += i
        end
      end
      i += 1
    end

    sieve
  end
end

def self.prime?(n)
  esieve[n] == 0
end

gets.to_i.times do
  n = gets.to_i
  total = 1
  skip_hash = { }
  i = 2
  while i <= n do
    if !skip_hash[i] && prime?(i)
      j = i
      while j <= n do
        skip_hash[j] = 1
        j *= i
      end
      j /= i
      total *= j
    end
    i += 1
  end
  puts total
end

File: projecteuler/Euler005.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {
  public static final int MAX = 41;

  public static int[] esieve() {
    int[] sieve = new int[MAX];
    sieve[0] = 1;
    sieve[1] = 1;

    for (int i = 2; i < MAX; i++) {
      if (sieve[i] == 0) {
        for (int j = i * i; j < MAX; j += i) {
          sieve[j] = 1;
        }
      }
    }

    return sieve;
  }

  public static void main(String[] args) {
    int total;
    int[] sieve = esieve();
    Scanner scanner = new Scanner(System.in);
    int t = scanner.nextInt();
    int n;
    int j;
    for(int k = 0; k < t; k++) {
      total = 1;
      n = scanner.nextInt();
      for(int i = 2; i <= n; i++) {
        if (sieve[i] == 0) {
          for (j = i; j <= n;) {
            j *= i;
          }
          j /= i;
          total *= j;
        }
      }
      System.out.println(total);
    }
  }
}
